### ISSUE TITLE
Nullable analysis doesn't learn from recursive patterns applied to conditional accesses

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#53308]
**Version Used**: 265a9c331231a042ffeb06a5a64058f90e82f369

[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4GIYFcA2OLA4CmABETAcQLABQtAsgIwAU2eAlANwMBMruOLrVoABACwkmzAEIB+EsHa0A3rRLqSUAGYlmwTQGcSykgGEQxkgAUATgHsADhZMBfEm5dKaG42p8bgADpTQNtHQIAVOwBlABcbWABzZi4SAHo0khg7EgB3CBsYJL91F1oyuhpxST45BS9Vbw1tXWBZYMNLMKdLDy8fRv8A4ND7B0iY+KSUznTM/MLiptLy4SqeEmkVEpIRAGYzeVNLRKJY2YMz2YqK0Q3TbeXdg5FGAAZ5bpOrkkvz93KQA)

```cs
#nullable enable

M1(null);
M2(null);

void M1(B? b)
{
    if (b is { C: { Prop: { } } })
    {
        b.C.Prop.ToString(); // no warning
    }
}

void M2(B? b)
{
    if (b?.C is { Prop: { } })
    {
        b.C.Prop.ToString(); // warning
    }
}

class B
{
    public C? C { get; set; }
}

class C
{
    public string? Prop { get; set; }
}
```

**Expected Behavior**: No warning in `M1` or `M2`

**Actual Behavior**: No warning in `M1`, but a warning in `M2`
